opt/panicbutt/plugins/reddit.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/python
+# =======================================
+#
+#  File Name : reddit.py
+#
+#  Purpose : take r/subreddit and make it a clickable link
+#
+#  Creation Date : 12-02-2018
+#
+#  Last Modified : Mon 12 February 2018 10:31:27 AM EST
+#
+#  Created By : Edward Olsen
+#
+# ========================================
+
+import slackbot.bot
+import re
+
+REDDIT_REGEX = "((\/r\/)|(r\/))(\w+)"
```

Comment:
I wonder if we want to pad the token we're looking for with spaces, like
`\s+((\/r\/)|(r\/))(\w+)\s+` to try and reduce false positives/trolling.

Unless that's the point.

Reply:
Good catch though I think the pattern \s?+ would work better. There might not always be a space before the match.

Reply:
Doesn't \s* do what you want?

Reply:
The regex tester I'm using doesn't like the \s*. Seems to throw off the parenthesis in the first group.

Reply:
```
>>> REDDIT_REGEX = r'((\/r\/)|(r\/))(\w+)'
>>> s = 'find me a subreddit'
>>> t = 'find me r/trees'
>>> r = 'find me buttsr/trees'
>>> q = 'find me lolbutts/r/trees'
>>> for i in [s,t,r,q]:
...     if re.search(REDDIT_REGEX, i, re.I):
...         print('Pattern matches "%s"' % i)
...
Pattern matches "find me r/trees"
Pattern matches "find me buttsr/trees"
Pattern matches "find me lolbutts/r/trees"
>>> NEW_REGEX = r'\s*((\/r\/)|(r\/))(\w+)'
>>> for i in [s,t,r,q]:
...     if re.search(NEW_REGEX, i, re.I):
...         print('Pattern matches "%s"' % i)
...
Pattern matches "find me r/trees"
Pattern matches "find me buttsr/trees"
Pattern matches "find me lolbutts/r/trees"
>>>
>>> OTHER_REGEX = r'\s?+((\/r\/)|(r\/))(\w+)'
>>> for i in [s,t,r,q]:
...     if re.search(NEW_REGEX, i, re.I):
...         print('Pattern matches "%s"' % i)
...
Pattern matches "find me r/trees"
Pattern matches "find me buttsr/trees"
Pattern matches "find me lolbutts/r/trees"
>>>
```

So with `\s*` and `\s?+`, I'm not convinced it's doing what we want; maybe it's acceptable to match `buttsr/trees` or `reddit.com/r/trees` but that wasn't my interpretation of what this plugin was for, which was a short-subreddit expander/linker.

Just my two cents.